api/filters/namespace/namespace.go

```diff
@@ -107,12 +123,11 @@ func (ns Filter) metaNamespaceHack(obj *yaml.RNode, gvk resid.Gvk) error {
 // - name: "something-else" # this will not have the namespace set
 //   ...
 func (ns Filter) roleBindingHack(obj *yaml.RNode, gvk resid.Gvk) error {
-	if gvk.Kind != roleBindingKind && gvk.Kind != clusterRoleBindingKind {
+	if !isRoleBinding(gvk.Kind) {
 		return nil
 	}
 
 	// Lookup the namespace field on all elements.
-	// We should change the fieldspec so this isn't necessary.
```

Comment:
Unless I'm missing something, it's impossible for fieldspecs to drive conditional logic like what this function is doing.